src/server/pps/server/s3g_sidecar_test.go

```diff
@@ -837,3 +839,57 @@ func TestS3SkippedDatums(t *testing.T) {
 		testS3SkippedDatums(t, c, userToken, ns, projectName, func(p, r string) string { return r + "." + p })
 	})
 }
+
+// TestDontDownloadData tests that when a pipeline sets `S3: true` on an input,
+// the worker binary doesn't download any datums from that input to the worker.
+// This test limits the size of the worker pod's disk to a smaller size than the
+// input data exposed via a `S3: true` input. When the worker binary attempts to
+// download all 10GB of input, the test times out.
+func TestDontDownloadData(t *testing.T) {
```

Comment:
I am a bit concerned about adding a test with 10GB of data. Would an existence check of smaller files not be sufficient?

Reply:
I reduced it to 2GB of data for now. Just duplicating a slack question here: would it make sense for the test to check for data in `.scratch` directly? Or would that be too implementation-dependent?

Reply:
Yeah, in the case of this bug the /pfs/input symlinks never get created, but we do still download all of the data, so the test has to either download enough data to blow up the machine (perhaps with the help of k8s disk limits), or it has to know "too much" about how the worker works today.

Reply:
Bit of an update: I may actually be required to scan `/pfs/.scratch`, as applying a storage limit to the container doesn't appear to do what we want. I tried reducing the amount of data written from 10GB to 2GB (which should still exceed the limit) and the test starting passing. The issue is that even though the container's ephemeral storage is limited to 1GB, `/pfs` is an `EmptyDir` mount that is unlimited:
```
...
Containers:
  user:
    ...
    Limits:
      ephemeral-storage:  1G
    Requests:
      ephemeral-storage:  1G
    ...
    Mounts:
      ...
      /pfs from pachyderm-worker (rw)
  ...
Volumes:
  ...
  pachyderm-worker:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
```

Previously, the test was just timing out because of the data size. I don't think there's a way to cap the size of the `pachyderm-worker` volume, so the `.scratch` approach it is?

Reply:
Test updated